SharedUtils/src/main/java/com/braintreepayments/api/BraintreeSharedPreferences.java

```diff
@@ -2,53 +2,31 @@
 
 import android.content.Context;
 import android.content.SharedPreferences;
-import android.os.Parcel;
-import android.os.Parcelable;
-import android.util.Base64;
 
-class BraintreeSharedPreferences {
-
-    static SharedPreferences getSharedPreferences(Context context) {
-        return context.getApplicationContext().getSharedPreferences("BraintreeApi", Context.MODE_PRIVATE);
-    }
-
-    static void putParcelable(Context context, String key, Parcelable parcelable) {
-        Parcel parcel = Parcel.obtain();
-        parcelable.writeToParcel(parcel, 0);
-        getSharedPreferences(context).edit()
-                .putString(key, Base64.encodeToString(parcel.marshall(), 0))
-                .apply();
-    }
+import androidx.security.crypto.EncryptedSharedPreferences;
+import androidx.security.crypto.MasterKey;
 
-    static Parcel getParcelable(Context context, String key) {
-        try {
-            byte[] requestBytes = Base64.decode(getSharedPreferences(context).getString(key, ""), 0);
-            Parcel parcel = Parcel.obtain();
-            parcel.unmarshall(requestBytes, 0, requestBytes.length);
-            parcel.setDataPosition(0);
+import java.io.IOException;
+import java.security.GeneralSecurityException;
 
-            return parcel;
-        } catch (Exception ignored) {}
-
-        return null;
-    }
+class BraintreeSharedPreferences {
 
-    static void remove(Context context, String key) {
-        getSharedPreferences(context)
-                .edit()
-                .remove(key)
-                .apply();
+    SharedPreferences getSharedPreferences(Context context) throws GeneralSecurityException, IOException {
+        return getSharedPreferences(context, "BraintreeApi");
     }
 
-    static void putString(Context context, String key, String value) {
```

Comment:
qq: Is there a way we could encapsulate `EncryptedSharedPreferences` within `BraintreeSharedPreferences` and keep all of the `getX()` and `putX()` methods?

Reply:
Good idea - I ended up encapsulating all the access to shared prefs. Most of the put/get methods are only used once each, but makes the testing story cleaner

Reply:
Yeah after seeing this PR I realize there are a lot of static classes related to SharedPrefs work. I honestly feel like `BraintreeSharedPreferences` makes sense as a singleton since shared prefs is a single shared persistent store.

Reply:
A few of the other classes could probably be moved to be non-static too. That feels like some of the `v4` work we wanted to do, but we punted on so we wouldn't delay the release.

Reply:
Would a static `getInstance()` cause the same test problems from the PR description?

Reply:
Yeah the static methods always throw a wrench in the tests. We have done constructor level injection in `@VisibleForTesting` methods in some places though. Some of our `package-private` classes still have a bit of a `v3` style from when we used Powermock everywhere.